### DISCUSSION TITLE
Cant open PID File, SSL_read: I/O error in xrdp Ubuntu 22.04.3 LTS

### DISCUSSION THREAD

**Discussion by rmcg3** [neutrinolabs/xrdp#2931] [Q&A]
Hi, I know there's a few questions about these issues and I've tried a lot of solutions and none of them seem to work for me.
I am running Ubuntu 22.04.3 LTS (GNU/Linux 5.16.9-051609-generic x86_64) and trying to xrdp into it from a Windows 10 machine.

Upon starting xrdp service I get:

> systemd[1]: Starting xrdp daemon...
>  xrdp[11261]: [INFO ] address [0.0.0.0] port [3389] mode 1
>  xrdp[11261]: [INFO ] listening to port 3389 on 0.0.0.0
>  xrdp[11261]: [INFO ] xrdp_listen_pp done
>  systemd[1]: xrdp.service: Can't open PID file /run/xrdp/xrdp.pid (yet?) after start: Operation not permitted
>  systemd[1]: Started xrdp daemon.
>  xrdp[11262]: [INFO ] starting xrdp with pid 11262
>  xrdp[11262]: [INFO ] address [0.0.0.0] port [3389] mode 1
>  xrdp[11262]: [INFO ] listening to port 3389 on 0.0.0.0
> 1 xrdp[11262]: [INFO ] xrdp_listen_pp done
> 

The `xrdp.log` reads:

> [20240131-17:25:52] [ERROR] SSL_read: I/O error
> [20240131-17:25:52] [ERROR] libxrdp_force_read: header read error
> [20240131-17:25:52] [ERROR] Processing [ITU-T T.125] Connect-Initial failed
> [20240131-17:25:52] [ERROR] [MCS Connection Sequence] receive connection request failed
> [20240131-17:25:52] [ERROR] xrdp_sec_incoming: xrdp_mcs_incoming failed
> [20240131-17:25:52] [ERROR] xrdp_rdp_incoming: xrdp_sec_incoming failed
> [20240131-17:25:52] [ERROR] xrdp_process_main_loop: libxrdp_process_incoming failed
> [20240131-17:25:52] [ERROR] xrdp_iso_send: trans_write_copy_s failed
> [20240131-17:25:52] [ERROR] Sending [ITU T.125] DisconnectProviderUltimatum failed
> [20240131-17:25:52] [INFO ] Socket 12: AF_INET6 connection received from ::ffff:172.20.104.50 port 58034
> [20240131-17:25:52] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
> [20240131-17:25:52] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
> [20240131-17:25:52] [INFO ] Connected client computer name: PHL-LPT-6CVXYW3
> [20240131-17:25:52] [WARN ] Received [MS-RDPBCGR] TS_UD_HEADER type 0xc006 is unknown (ignored)
> [20240131-17:25:52] [WARN ] Received [MS-RDPBCGR] TS_UD_HEADER type 0xc00a is unknown (ignored)
> [20240131-17:25:52] [INFO ] xrdp_load_keyboard_layout: Keyboard information sent by the RDP client, keyboard_type:[0x07], keyboard_subtype:[0x00], keylayout:[0x0000>
> [20240131-17:25:52] [INFO ] xrdp_load_keyboard_layout: model [] variant [] layout [us] options []
> [20240131-17:25:52] [INFO ] TLS connection established from ::ffff:172.20.104.50 port 58034: TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384
> [20240131-17:25:52] [INFO ] xrdp_caps_process_pointer: client supports new(color) cursor
> [20240131-17:25:52] [INFO ] xrdp_process_offscreen_bmpcache: support level 1 cache size 10485760 MB cache entries 100
> [20240131-17:25:52] [INFO ] xrdp_caps_process_codecs: nscodec, codec id 1, properties len 3
> [20240131-17:25:52] [WARN ] xrdp_caps_process_codecs: unknown codec id 5
> [20240131-17:25:52] [INFO ] xrdp_caps_process_codecs: RemoteFX, codec id 3, properties len 49
> [20240131-17:25:52] [INFO ] Loading keymap file /etc/xrdp/km-00000409.ini
> [20240131-17:25:52] [WARN ] local keymap file for 0x00000409 found and doesn't match built in keymap, using local keymap file
> [20240131-17:25:59] [INFO ] connecting to sesman ip 127.0.0.1 port 3350
> [20240131-17:25:59] [INFO ] xrdp_wm_log_msg: sesman connect ok
> [20240131-17:25:59] [INFO ] sesman connect ok
> [20240131-17:25:59] [INFO ] sending login info to session manager, please wait...
> [20240131-17:25:59] [INFO ] xrdp_wm_log_msg: login successful for display 11
> [20240131-17:25:59] [INFO ] login successful for display 11
> [20240131-17:25:59] [INFO ] loaded module 'libxup.so' ok, interface size 10296, version 4
> [20240131-17:25:59] [INFO ] started connecting
> [20240131-17:25:59] [INFO ] lib_mod_connect: connecting via UNIX socket
> [20240131-17:25:59] [INFO ] lib_mod_log_peer: xrdp_pid=11270 connected to X11rdp_pid=11274 X11rdp_uid=1000 X11rdp_gid=1000 client_ip=::ffff:172.20.104.50 client_por>
> [20240131-17:25:59] [INFO ] connected ok

It seems like it successfully connects me here but I get the "Oh no! Something has gone wrong. ..." error in the RDP client on my Windows host machine.

I've tried the whole as root & w/ sudo etc..:

> chmod 644 /etc/xrdp/cert.pem
> chmod 640 /etc/xrdp/key.pem
> chown :xrdp /etc/xrdp/key.pem
> 

and when I try to change the ownership to the key file I still get:

> lrwxrwxrwx 1 root root 36 Jan 31 16:47 /etc/xrdp/cert.pem -> /etc/ssl/certs/ssl-cert-snakeoil.pem
> lrwxrwxrwx 1 root root 38 Jan 31 16:47 /etc/xrdp/key.pem -> /etc/ssl/private/ssl-cert-snakeoil.key

after running `ls -l /etc/xrdp/key.pem /etc/xrdp/cert.pem`

Thanks,

**Comment by matt335672** [comment]
The "Oh no! Something has gone wrong. ..." error generally means that the X server isn't supporting 3D acceleration. There's nothing wrong with your certificate.

Which backend are you using? Xorg or Xvnc?

**Reply by rmcg3** [reply]
I am using Xorg

**Reply by matt335672** [reply]
You're not logged in graphically to the same machine console using the same account are you?

If not, please post ~/.xorgxrdp.11.log

**Reply by rmcg3** [reply]
.xorgxrdp.11.log (some personal info removed near top):

> [  5775.607] 
> X.Org X Server 1.21.1.4
> X Protocol Version 11, Revision 0
> [  5775.607] Current Operating System: Linux  5.16.9-051609-generic SMP PREEMPT Fri Feb 11 09:59:18 UTC 2022 x86_64
> [  5775.607] Kernel command line: BOOT_IMAGE=/boot/vmlinuz-5.16.9-051609-generic  ro quiet splash zswap.enabled=1 zswap.zpool=z3fold zswap.compressor=lz4
> [  5775.607] xorg-server 2:21.1.4-2ubuntu1.7~22.04.8 (For technical support please see http://www.ubuntu.com/support) 
> [  5775.607] Current version of pixman: 0.40.0
> [  5775.607] 	Before reporting problems, check http://wiki.x.org
> 	to make sure that you have the latest version.
> [  5775.607] Markers: (--) probed, (**) from config file, (==) default setting,
> 	(++) from command line, (!!) notice, (II) informational,
> 	(WW) warning, (EE) error, (NI) not implemented, (??) unknown.
> [  5775.607] (++) Log file: ".xorgxrdp.11.log", Time: Wed Jan 31 17:33:05 2024
> [  5775.607] (++) Using config file: "/etc/X11/xrdp/xorg.conf"
> [  5775.607] (==) Using system config directory "/usr/share/X11/xorg.conf.d"
> [  5775.607] (**) Option "defaultserverlayout" "X11 Server"
> [  5775.607] (**) ServerLayout "X11 Server"
> [  5775.607] (**) |-->Screen "Screen (xrdpdev)" (0)
> [  5775.607] (**) |   |-->Monitor "Monitor"
> [  5775.607] (**) |   |-->Device "Video Card (xrdpdev)"
> [  5775.607] (**) |-->Input Device "xrdpMouse"
> [  5775.607] (**) |-->Input Device "xrdpKeyboard"
> [  5775.607] (**) Option "DontVTSwitch" "on"
> [  5775.607] (**) Option "AutoAddDevices" "off"
> [  5775.607] (**) Not automatically adding devices
> [  5775.607] (==) Automatically enabling devices
> [  5775.607] (==) Automatically adding GPU devices
> [  5775.607] (==) Automatically binding GPU devices
> [  5775.607] (==) Max clients allowed: 256, resource mask: 0x1fffff
> [  5775.608] (WW) The directory "/usr/share/fonts/X11/cyrillic" does not exist.
> [  5775.608] 	Entry deleted from font path.
> [  5775.608] (WW) The directory "/usr/share/fonts/X11/100dpi/" does not exist.
> [  5775.608] 	Entry deleted from font path.
> [  5775.608] (WW) The directory "/usr/share/fonts/X11/75dpi/" does not exist.
> [  5775.608] 	Entry deleted from font path.
> [  5775.608] (WW) The directory "/usr/share/fonts/X11/100dpi" does not exist.
> [  5775.608] 	Entry deleted from font path.
> [  5775.608] (WW) The directory "/usr/share/fonts/X11/75dpi" does not exist.
> [  5775.608] 	Entry deleted from font path.
> [  5775.608] (==) FontPath set to:
> 	/usr/share/fonts/X11/misc,
> 	/usr/share/fonts/X11/Type1,
> 	built-ins
> [  5775.608] (==) ModulePath set to "/usr/lib/xorg/modules"
> [  5775.608] (II) Loader magic: 0x55c184f8a020
> [  5775.608] (II) Module ABI versions:
> [  5775.608] 	X.Org ANSI C Emulation: 0.4
> [  5775.608] 	X.Org Video Driver: 25.2
> [  5775.608] 	X.Org XInput driver : 24.4
> [  5775.608] 	X.Org Server Extension : 10.0
> [  5775.609] (II) systemd-logind: took control of session /org/freedesktop/login1/session/c7
> [  5775.610] (II) xfree86: Adding drm device (/dev/dri/card0)
> [  5775.610] (II) Platform probe for /sys/devices/pci0000:00/0000:00:0f.0/drm/card0
> [  5775.611] (EE) systemd-logind: failed to take device /dev/dri/card0: Operation not permitted
> [  5775.615] (--) PCI:*(0@0:15:0) 15ad:0405:15ad:0405 rev 0, Mem @ 0xe8000000/134217728, 0xfe000000/8388608, I/O @ 0x00001070/16, BIOS @ 0x????????/131072
> [  5775.615] (II) "glx" will be loaded. This was enabled by default and also specified in the config file.
> [  5775.615] (II) LoadModule: "dbe"
> [  5775.615] (II) Module "dbe" already built-in
> [  5775.615] (II) LoadModule: "ddc"
> [  5775.615] (II) Module "ddc" already built-in
> [  5775.615] (II) LoadModule: "extmod"
> [  5775.615] (II) Module "extmod" already built-in
> [  5775.615] (II) LoadModule: "glx"
> [  5775.616] (II) Loading /usr/lib/xorg/modules/extensions/libglx.so
> [  5775.621] (II) Module glx: vendor="X.Org Foundation"
> [  5775.621] 	compiled for 1.21.1.4, module version = 1.0.0
> [  5775.621] 	ABI class: X.Org Server Extension, version 10.0
> [  5775.621] (II) LoadModule: "int10"
> [  5775.621] (II) Loading /usr/lib/xorg/modules/libint10.so
> [  5775.621] (II) Module int10: vendor="X.Org Foundation"
> [  5775.622] 	compiled for 1.21.1.4, module version = 1.0.0
> [  5775.622] 	ABI class: X.Org Video Driver, version 25.2
> [  5775.622] (II) LoadModule: "record"
> [  5775.622] (II) Module "record" already built-in
> [  5775.622] (II) LoadModule: "vbe"
> [  5775.622] (II) Loading /usr/lib/xorg/modules/libint10.so
> [  5775.622] (II) Module int10: vendor="X.Org Foundation"
> [  5775.622] 	compiled for 1.21.1.4, module version = 1.0.0
> [  5775.622] 	ABI class: X.Org Video Driver, version 25.2
> [  5775.622] (II) LoadModule: "glamoregl"
> [  5775.622] (II) Loading /usr/lib/xorg/modules/libglamoregl.so
> [  5775.624] (II) Module glamoregl: vendor="X.Org Foundation"
> [  5775.625] 	compiled for 1.21.1.4, module version = 1.0.1
> [  5775.625] 	ABI class: X.Org ANSI C Emulation, version 0.4
> [  5775.625] (II) LoadModule: "xorgxrdp"
> [  5775.625] (II) Loading /usr/lib/xorg/modules/libxorgxrdp.so
> [  5775.625] (II) Module XORGXRDP: vendor="X.Org Foundation"
> [  5775.625] 	compiled for 1.21.1.3, module version = 0.2.17
> [  5775.625] 	ABI class: X.Org Video Driver, version 25.2
> [  5775.625] xorgxrdpSetup:
> [  5775.625] (II) LoadModule: "fb"
> [  5775.625] (II) Module "fb" already built-in
> [  5775.625] (II) LoadModule: "xrdpdev"
> [  5775.625] (II) Loading /usr/lib/xorg/modules/drivers/xrdpdev_drv.so
> [  5775.625] (II) Module XRDPDEV: vendor="X.Org Foundation"
> [  5775.625] 	compiled for 1.21.1.3, module version = 0.2.17
> [  5775.625] 	ABI class: X.Org Video Driver, version 25.2
> [  5775.625] xrdpdevSetup:
> [  5775.625] (II) LoadModule: "xrdpmouse"
> [  5775.625] (II) Loading /usr/lib/xorg/modules/input/xrdpmouse_drv.so
> [  5775.625] (II) Module XRDPMOUSE: vendor="X.Org Foundation"
> [  5775.625] 	compiled for 1.21.1.3, module version = 0.2.17
> [  5775.625] 	Module class: X.Org XInput Driver
> [  5775.625] 	ABI class: X.Org XInput driver, version 24.4
> [  5775.625] rdpmousePlug:
> [  5775.625] (II) LoadModule: "xrdpkeyb"
> [  5775.625] (II) Loading /usr/lib/xorg/modules/input/xrdpkeyb_drv.so
> [  5775.625] (II) Module XRDPKEYB: vendor="X.Org Foundation"
> [  5775.625] 	compiled for 1.21.1.3, module version = 0.2.17
> [  5775.625] 	Module class: X.Org XInput Driver
> [  5775.625] 	ABI class: X.Org XInput driver, version 24.4
> [  5775.625] rdpkeybPlug:
> [  5775.625] rdpIdentify:
> [  5775.625] (II) XRDPDEV: driver for xrdp: XRDPDEV
> [  5775.625] rdpDriverFunc: op 10
> [  5775.625] (WW) Falling back to old probe method for XRDPDEV
> [  5775.625] rdpProbe:
> [  5775.625] (II) Loading sub module "fb"
> [  5775.625] (II) LoadModule: "fb"
> [  5775.625] (II) Module "fb" already built-in
> [  5775.625] rdpProbe: found DRMDevice xorg.conf value [/dev/dri/renderD128]
> [  5775.625] rdpProbe: found DRI3 xorg.conf value [1]
> [  5775.625] (II) XRDPDEV(0): using default device
> [  5775.625] (WW) VGA arbiter: cannot open kernel arbiter, no multi-card support
> [  5775.625] rdpPreInit:
> [  5775.625] rdpPreInit: /dev/dri/renderD128 open failed
> [  5775.625] (**) XRDPDEV(0): Depth 24, (--) framebuffer bpp 32
> [  5775.625] (==) XRDPDEV(0): RGB weight 888
> [  5775.625] (==) XRDPDEV(0): Using gamma correction (1.0, 1.0, 1.0)
> [  5775.625] (==) XRDPDEV(0): Default visual is TrueColor
> [  5775.625] (==) XRDPDEV(0): DPI set to (96, 96)
> [  5775.625] (II) XRDPDEV(0): 	mode "640x480" ok
> [  5775.625] (II) XRDPDEV(0): 	mode "800x600" ok
> [  5775.625] (II) XRDPDEV(0): Virtual size is 800x600 (pitch 800)
> [  5775.625] (**) XRDPDEV(0):  Default mode "800x600": 36.0 MHz (scaled from 0.0 MHz), 35.2 kHz, 56.2 Hz
> [  5775.625] (II) XRDPDEV(0): Modeline "800x600"x0.0   36.00  800 824 896 1024  600 601 603 625 +hsync +vsync (35.2 kHz d)
> [  5775.625] rdpScreenInit: virtualX 800 virtualY 600 rgbBits 8 depth 24
> [  5775.625] rdpScreenInit: pfbMemory bytes 1920000
> [  5775.625] rdpScreenInit: pfbMemory 0x7fa70acee010
> [  5775.626] rdpSimdInit: assigning yuv functions
> [  5775.626] rdpSimdInit: cpuid ax 1 cx 0 return ax 0x00050654 bx 0x03040800 cx 0xfffa3203 dx 0x1fabfbff
> [  5775.626] rdpSimdInit: sse2 amd64 yuv functions assigned
> [  5775.626] (==) XRDPDEV(0): Backing store enabled
> [  5775.626] rdpClientConInit: disconnect idle session after [0] sec
> [  5775.626] rdpClientConInit: kill disconnected [0] timeout [0] sec
> [  5775.626] rdpXvInit: depth 24
> [  5775.626] rdpScreenInit: out
> [  5775.626] (II) Initializing extension Generic Event Extension
> [  5775.626] (II) Initializing extension SHAPE
> [  5775.626] (II) Initializing extension MIT-SHM
> [  5775.626] (II) Initializing extension XInputExtension
> [  5775.626] (II) Initializing extension XTEST
> [  5775.626] (II) Initializing extension BIG-REQUESTS
> [  5775.626] (II) Initializing extension SYNC
> [  5775.626] (II) Initializing extension XKEYBOARD
> [  5775.626] (II) Initializing extension XC-MISC
> [  5775.627] (II) Initializing extension SECURITY
> [  5775.627] (II) Initializing extension XFIXES
> [  5775.627] (II) Initializing extension RENDER
> [  5775.627] (II) Initializing extension RANDR
> [  5775.627] (II) Initializing extension COMPOSITE
> [  5775.627] (II) Initializing extension DAMAGE
> [  5775.627] (II) Initializing extension MIT-SCREEN-SAVER
> [  5775.627] (II) Initializing extension DOUBLE-BUFFER
> [  5775.627] (II) Initializing extension RECORD
> [  5775.628] (II) Initializing extension DPMS
> [  5775.628] (II) Initializing extension Present
> [  5775.628] (II) Initializing extension DRI3
> [  5775.628] (II) Initializing extension X-Resource
> [  5775.628] (II) Initializing extension XVideo
> [  5775.628] (II) Initializing extension XVideo-MotionCompensation
> [  5775.628] (II) Initializing extension SELinux
> [  5775.628] (II) SELinux: Disabled on system
> [  5775.628] (II) Initializing extension GLX
> [  5775.628] (II) AIGLX: Screen 0 is not DRI2 capable
> [  5775.661] (II) IGLX: Loaded and initialized swrast
> [  5775.661] (II) GLX: Initialized DRISWRAST GL provider for screen 0
> [  5775.661] (II) Initializing extension XFree86-VidModeExtension
> [  5775.661] (II) Initializing extension XFree86-DGA
> [  5775.661] (II) Initializing extension XFree86-DRI
> [  5775.661] (II) Initializing extension DRI2
> [  5775.661] rdpCreateScreenResources:
> [  5775.683] (II) Using input driver 'XRDPMOUSE' for 'xrdpMouse'
> [  5775.683] (**) Option "CorePointer"
> [  5775.683] (**) xrdpMouse: always reports core events
> [  5775.683] rdpmousePreInit: drv 0x55c1850037a0 info 0x55c1852fd860, flags 0x0
> [  5775.683] (II) XINPUT: Adding extended input device "xrdpMouse" (type: Mouse, id 6)
> [  5775.683] rdpmouseControl: what 0
> [  5775.683] rdpmouseDeviceInit:
> [  5775.683] rdpmouseCtrl:
> [  5775.683] rdpRegisterInputCallback: type 1 proc 0x7fa70b8413b0
> [  5775.683] (**) xrdpMouse: (accel) keeping acceleration scheme 1
> [  5775.683] (**) xrdpMouse: (accel) acceleration profile 0
> [  5775.683] (**) xrdpMouse: (accel) acceleration factor: 2.000
> [  5775.683] (**) xrdpMouse: (accel) acceleration threshold: 4
> [  5775.683] rdpmouseControl: what 1
> [  5775.684] rdpmouseDeviceOn:
> [  5775.684] (II) Using input driver 'XRDPKEYB' for 'xrdpKeyboard'
> [  5775.684] (**) Option "CoreKeyboard"
> [  5775.684] (**) xrdpKeyboard: always reports core events
> [  5775.684] rdpkeybPreInit: drv 0x55c185003750 info 0x55c1853009e0, flags 0x0
> [  5775.684] (II) XINPUT: Adding extended input device "xrdpKeyboard" (type: Keyboard, id 7)
> [  5775.684] rdpkeybControl: what 0
> [  5775.684] rdpkeybDeviceInit:
> [  5775.698] rdpkeybChangeKeyboardControl:
> [  5775.698] rdpkeybChangeKeyboardControl: autoRepeat on
> [  5775.698] rdpRegisterInputCallback: type 0 proc 0x7fa70b80ed50
> [  5775.698] rdpkeybControl: what 1
> [  5775.698] rdpkeybDeviceOn:
> [  5775.698] rdpkeybChangeKeyboardControl:
> [  5775.698] rdpkeybChangeKeyboardControl: autoRepeat on
> [  5775.708] (II) config/udev: Adding input device Power Button (/dev/input/event0)
> [  5775.708] (II) AutoAddDevices is off - not adding device.
> [  5775.708] (II) config/udev: Adding drm device (/dev/dri/card0)
> [  5775.708] (II) xfree86: Adding drm device (/dev/dri/card0)
> [  5775.708] (II) Platform probe for /sys/devices/pci0000:00/0000:00:0f.0/drm/card0
> [  5775.709] (EE) systemd-logind: failed to take device /dev/dri/card0: Operation not permitted
> [  5775.710] (II) config/udev: Adding input device AT Translated Set 2 keyboard (/dev/input/event1)
> [  5775.710] (II) AutoAddDevices is off - not adding device.
> [  5775.710] (II) config/udev: Adding input device VirtualPS/2 VMware VMMouse (/dev/input/event3)
> [  5775.710] (II) AutoAddDevices is off - not adding device.
> [  5775.710] (II) config/udev: Adding input device VirtualPS/2 VMware VMMouse (/dev/input/mouse1)
> [  5775.710] (II) AutoAddDevices is off - not adding device.
> [  5775.711] (II) config/udev: Adding input device VirtualPS/2 VMware VMMouse (/dev/input/event2)
> [  5775.711] (II) AutoAddDevices is off - not adding device.
> [  5775.711] (II) config/udev: Adding input device VirtualPS/2 VMware VMMouse (/dev/input/mouse0)
> [  5775.711] (II) AutoAddDevices is off - not adding device.
> [  5775.717] rdpDeferredRandR:
> [  5775.717] rdpResizeSession: width 1024 height 768
> [  5775.717]   calling RRScreenSizeSet
> [  5775.717] rdpRRScreenSetSize: width 1024 height 768 mmWidth 271 mmHeight 203
> [  5775.717] rdpRRGetInfo:
> [  5775.717]   screen resized to 1024x768
> [  5775.718]   RRScreenSizeSet ok 1
> [  5775.718] rdpResizeSession: width 1920 height 1080
> [  5775.718]   calling RRScreenSizeSet
> [  5775.718] rdpRRScreenSetSize: width 1920 height 1080 mmWidth 508 mmHeight 286
> [  5775.719] rdpRRGetInfo:
> [  5775.719]   screen resized to 1920x1080
> [  5775.723]   RRScreenSizeSet ok 1
> [  5775.723] rdpClientConGotConnection:
> [  5775.723] rdpClientConGotConnection: g_sck_accept ok new_sck 9
> [  5775.723] rdpClientConGetConnection: idle_disconnect_timeout set to non-positive value, idle timer turned off
> [  5775.723] rdpAddClientConToDev: adding first clientCon 0x55c18532b810
> [  5775.723] rdpClientConProcessMsgVersion: version 0 0 0 1
> [  5775.723] rdpClientConProcessScreenSizeMsg: set width 1920 height 1080 bpp 32
> [  5775.723] rdpClientConProcessScreenSizeMsg: shmemid 48 shmemptr 0x7fa6fe2f1000
> [  5775.723] rdpClientConProcessMsgClientInput: invalidate x 0 y 0 cx 1920 cy 1080
> [  5775.723] KbdSync: toggling num lock
> [  5775.723] rdpkeybChangeKeyboardControl:
> [  5775.723] rdpkeybChangeKeyboardControl: autoRepeat on
> [  5775.798] rdpInDeferredRepeatCallback:
> [  5775.798] rdpkeybChangeKeyboardControl:
> [  5775.798] rdpkeybChangeKeyboardControl: autoRepeat off
> [  5775.798] rdpInDeferredRepeatCallback:
> [  5775.798] rdpkeybChangeKeyboardControl:
> [  5775.798] rdpkeybChangeKeyboardControl: autoRepeat off
> [  5775.823] rdpInDeferredRepeatCallback:
> [  5775.823] rdpkeybChangeKeyboardControl:
> [  5775.823] rdpkeybChangeKeyboardControl: autoRepeat off
> [  5775.896] rdpClientConProcessMsgClientInfo:
> [  5775.896]   got client info bytes 7072
> [  5775.896]   jpeg support 0
> [  5775.896]   offscreen support 1
> [  5775.896]   offscreen size 10485760
> [  5775.896]   offscreen entries 100
> [  5775.896]   client can not do offscreen to offscreen blits
> [  5775.896]   client can do new(color) cursor
> [  5775.896]   client can not do multimon
> [  5775.896] rdpRRSetRdpOutputs: numCrtcs 0 numOutputs 0 monitorCount 0
> [  5775.896] rdpRRSetRdpOutputs: add output 0 left 0 top 0 width 1920 height 1080
> [  5775.896] rdpLoadLayout: keylayout 0x00000409 variant  display 11
> [  5775.896] rdpkeybChangeKeyboardControl:
> [  5775.896] rdpkeybChangeKeyboardControl: autoRepeat on
> [  5775.896] rdpkeybChangeKeyboardControl:
> [  5775.896] rdpkeybChangeKeyboardControl: autoRepeat on
> [  5775.996] rdpInDeferredRepeatCallback:
> [  5775.996] rdpkeybChangeKeyboardControl:
> [  5775.996] rdpkeybChangeKeyboardControl: autoRepeat off
> [  5775.996] rdpInDeferredRepeatCallback:
> [  5775.996] rdpkeybChangeKeyboardControl:
> [  5775.996] rdpkeybChangeKeyboardControl: autoRepeat off
> [  5779.991] KbdSync: toggling num lock
> [  5779.991] rdpkeybChangeKeyboardControl:
> [  5779.991] rdpkeybChangeKeyboardControl: autoRepeat off
> [  5779.991] rdpkeybChangeKeyboardControl:
> [  5779.991] rdpkeybChangeKeyboardControl: autoRepeat off
> [  6468.347] rdpClientConRecv: g_sck_recv failed(returned 0)
> [  6468.347] rdpClientConRecvMsg: error
> [  6468.347] rdpClientConCheck: rdpClientConGotData failed
> [  6468.347] rdpClientConDisconnect:
> [  6468.347] rdpRemoveClientConFromDev: removing clientCon 0x55c18532b810

**Comment by matt335672** [comment]
That all looks reasonably OK too.

Does GNOME work on the machine console, or are you trying to set up headless access?
